dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/BaseIdentifiableObject.java

```diff
@@ -420,8 +420,11 @@ public void setFavorites(Set<String> favorites) {
   @JsonProperty
   @JacksonXmlProperty(namespace = DxfNamespaces.DXF_2_0)
   public boolean isFavorite() {
+    if (!CurrentUserUtil.hasCurrentUser()) {
+      return false;
+    }
     UserDetails user = CurrentUserUtil.getCurrentUserDetails();
-    return user != null && favorites != null && favorites.contains(user.getUid());
+    return favorites != null && favorites.contains(user.getUid());
```

Comment:
The `favorites != null` part should be lifted up as a condition for early exit. In that case we do not even have to access the current user.

Reply:
I mean make a `if (favorites == null) return false`